src/lib/util.ts

```diff
@@ -144,10 +144,51 @@ export function convertXPtoLVL(xp: number, cap = 99) {
 
 	return cap;
 }
+export function determineMiningTime(
+	quantity: number | null,
+	user: KlasaUser,
+	ore: Ore,
+	ticksBetweenRolls: number,
+	glovesRate: number,
+	armourEffect: number,
+	miningCapeEffect: number,
+	powerMining: boolean,
+	lvl: number
+): [number, number] {
+	let { intercept } = ore;
+	if (ore.id === 1625 && user.hasItemEquippedAnywhere('Amulet of glory')) {
+		intercept *= 3;
+	}
+	let timeElapsed = 0;
 
-export function determineScaledOreTime(xp: number, respawnTime: number, lvl: number) {
-	const t = xp / (lvl / 4 + 0.5) + ((100 - lvl) / 100 + 0.75);
-	return Math.floor((t + respawnTime) * 1000) * 1.2;
+	const bankTime = ore.bankingTime;
+	const chanceOfSuccess = ore.slope * lvl + intercept;
+	const respawnTimeOrPick = ticksBetweenRolls > ore.respawnTime ? ticksBetweenRolls : ore.respawnTime;
+
+	let newQuantity = 0;
+	while (timeElapsed < user.maxTripLength('Mining') / (Time.Second * 0.6)) {
```

Comment:
Okay so I have a lot of problems with this.

0. How can you mine just a few rocks with this? It forces a full trip down your throat every time.
1. This is extremely inefficient. While it does simulate well, this is going to be calculating 2000 iterations for a 40 minute trip. (3000/hr)
2. This returns the number of successful ore mined which is then displayed to the user, so the user will likely cancel until they get a high number of rocks, especially with things like Runite/Amethyst. 
3. These factors compound when a user cancels and retries multiple times, running thousands of iterations per attempt.

This needs to be thought out more. Probably the best way to handle 0+2 are:
- Do the 'fakeDuration' thing (don't forget the +m status message too)
- Accept quantities and let the loop go until maxTripLenth OR quantity specified is reached.

As for issues 1+3, I think we need to think of some optimizations, potentially at the cost of some accuracy, although might be possible to optimize without losing accuracy.

**Ooh here's a good idea** Maybe we can make a few fakeDuration mode where it says, "Trip will take between XX and YY minutes"  [obviously storing both for consistency with +m status]
And  the trip will finish sometime in the middle. 

Best way to achieve this? 
```ts
const [, realTime] = determineMiningTime(qty, etc);
const minTime = reduceNumByPercent(realTime, 25);
const maxTime = increaseNumByPercent(realTime, 25);
const fakeMin = addVariation(minTime, 20);
const fakeMax = addVariation(minTime, 20);
console.log(`Trip will take between ${formatDuration(fakeMin)} and ${formatDuration(fakeMax)}`);
```

This guarantees that the real time is between the min+max, and scales proportionally by trip length

Reply:
Found a solution based on information from this feedback.